packages/loader/container-loader/src/container.ts

```diff
@@ -1471,7 +1476,9 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
 
         const state = this._connectionState === ConnectionState.Connected;
-        this.context.setConnectionState(state, this.clientId);
+        if (!this.context.disposed) {
```

Comment:
Hard to say for sure, but it feels like that's the wrong place for this check. I.e. why would this method be called in the first place if context is disposed? It feels like it's either closed container (in which case we should not call into here), or we are transitioning between contexts (in which case we should call this API after process is done).

Reply:
just saw this puking in the test when closing the container in contextDisposed event

Reply:
Can you please look deeper into here? I feel like that IF should happen down the stack

Reply:
(node:33412) UnhandledPromiseRejectionWarning: Error: Runtime is closed
    at ContainerRuntime.verifyNotClosed (d:\code\fluid\packages\runtime\container-runtime\src\containerRuntime.ts:1940:19)
    at ContainerRuntime.setConnectionState (d:\code\fluid\packages\runtime\container-runtime\src\containerRuntime.ts:1072:14)
    at ContainerContext.setConnectionState (d:\code\fluid\packages\loader\container-loader\src\containerContext.ts:238:17)
    at Container.propagateConnectionState (d:\code\fluid\packages\loader\container-loader\src\container.ts:1480:22)
    at Container.setConnectionState (d:\code\fluid\packages\loader\container-loader\src\container.ts:1463:18)
    at DeltaManager.<anonymous> (d:\code\fluid\packages\loader\container-loader\src\container.ts:1331:18)
    at DeltaManager.emit (events.js:210:5)
    at DeltaManager.disconnectFromDeltaStream (d:\code\fluid\packages\loader\container-loader\src\deltaManager.ts:1092:14)
    at DeltaManager.close (d:\code\fluid\packages\loader\container-loader\src\deltaManager.ts:849:14)
    at Container.close (d:\code\fluid\packages\loader\container-loader\src\container.ts:494:28)
    at Container.<anonymous> (d:\code\fluid\packages\test\end-to-end-tests\src\test\codePropsal.spec.ts:179:27)
    at Object.onceWrapper (events.js:300:26)
    at Container.emit (events.js:215:7)
    at Container.emit (d:\code\fluid\packages\utils\telemetry-utils\src\eventEmitterWithErrorHandling.ts:25:28)
    at Container.reloadContextCore (d:\code\fluid\packages\loader\container-loader\src\container.ts:811:18)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processImmediate (internal/timers.js:412:9)

Reply:
that is the callstack. i just don't think we've even been able to close before while the context is disposed, as we previously always reloaded

Reply:
the crux of the issue is you can't set the connect state to disconnected on disposed context

Reply:
we could push it down one layer of the stack to ContainerContext.setConnectionState, but i think that should throw for accessing a disposed object, then we would still need to above check, otherwise it is just a slient failure which could cause issues